### ISSUE TITLE
Few issues!

### ISSUE THREAD

**Issue by Siyfion** [sergeyt/meteor-typeahead#61]
Firstly, great work with this wrapper, it's almost perfect for my use-case, which is always nice! :+1: 

Okay, so issues:

- The `data-selected` value takes a helper function, however inside that helper function `Template.instance()` no longer returns the template instance associated with the helper, it just returns null. This is really awkward when trying to set anything that exists on the template instance, which is quite a likely use-case!

- If you have two templates rendered at any one point, both with typeahead controls in them, calling `Meteor.typeahead.inject();` in both template's `rendered` functions causes some serious issues. However, not doing so means that each template cannot be used anywhere else, independently.

**Comment by sergeyt** [comment]
@Siyfion  re issue `Meteor.typeahead.inject()` - I think using `Template.instance()` should help to detect whether we need to apply typeahead behaviour only to one element, other fix - do not apply typeahead if it is already applied in previous `Meteor.typeahead.inject()` call. Please feel free to PR this issue.

**Comment by Siyfion** [comment]
If you can give me just a few pointers as to where in your code I need to be looking, etc. I'll try my best, although I'm still pretty new to Meteor generally, so I may not be able to help much!

**Comment by sergeyt** [comment]
@Siyfion possible steps to fix `Meteor.typeahead.inject()` issue:
* clone the repo
* run the demo app
* add add'l `Meteor.typeahead.inject()` call (i.e. call it twise) at meteor startup
* fix `Meteor.typeahead` to avoid double initialization of typeahead plugin for given DOM element, AFAIR $(e).data('ttTypeahead') should be null when typeahead is not initialized for given input element.

i.e. I think the fix of this issue should be one-liner.

To fix first issue I guess it is needed to bind event handler to `Template.instance()` context somehow and I don't know how to do that, usually looking into code of some built-in meteor package helps to find the common pattern or search stackoverflow, etc resources.

BTW I'm also newbie in meteor since I did not use it at all in my work, this package is mainly for learning purposes, but it seems the package works good enough.

**Comment by Siyfion** [comment]
Well, changing the inject function to check if there is a template instance set solves my main issue, as calling inject in a template's `rendered` function now only affects typeahead elements in the template.

However, if you call inject multiple times it still causes issues... Looking into that now.

```javascript
/**
 * Activates all typeahead elements.
 * @param selector (optional) selector to find typeahead elements to be activated
 */
Meteor.typeahead.inject = function(selector) {
	if (!selector) {
		selector = '.typeahead';
	}

	// See if we have a template instance to reference
	var template = Template.instance();
	if (!template) {
		// If we don't, just init on the entire DOM
		$(selector).each(init_typeahead);
	} else {
		// Otherwise just init this template's typeaheads
		template.$(selector).each(init_typeahead);
	}
};

function init_typeahead(index, element) {
	try {
		Meteor.typeahead(element);
	} catch (err) {
		console.log(err);
		return;
	}
}
```

**Comment by Siyfion** [comment]
And simply wrapping:

```javascript
if (element.data('ttTypeahead') == null) {
  Meteor.typeahead(element);
}
```

**Comment by Siyfion** [comment]
To set the context, it seems as though this bit of code:

```javascript
        // bind event handlers
	[
		"opened",
		"closed",
		"selected",
		"autocompleted",
	].forEach(function(name) {
		var fn = resolve_template_function($e[0], $e.data(name));
		if ($.isFunction(fn)) {
			instance.on('typeahead:' + name, fn);
		}
	});
```

Would have to somehow set the context of the helper function to that of the context of `Blaze.getData(instance)` or similar...?

**Comment by sergeyt** [comment]
@Siyfion I think you are on right way, please go ahead to make PR.

**Comment by Siyfion** [comment]
PR created in #62

**Comment by sergeyt** [comment]
@Siyfion please let me know if you need release with PR #62 without fix for first issue

**Comment by Siyfion** [comment]
@sergeyt Yes please! The first issue I haven't managed to figure out as-yet... So lets get this sorted first!

**Comment by Siyfion** [comment]
@sergeyt Can we get this version released to atmosphere?

**Comment by sergeyt** [comment]
@Siyfion I've published new version to atmosphere with the fix for second issue, also checked that demo app is stil working (found and fixed bug init_typeahead).

**Comment by Siyfion** [comment]
Brilliant! Let me give that a try... :+1:

**Comment by sergeyt** [comment]
@Siyfion the first issue with template instance should be fixed by changeset 101bc77, i.e. handler function is bound to template instance and you should be able to use `this` (reference to Blaze.Template) instead of `Template.instance()` that returns `null` in event handler context.

**Comment by Siyfion** [comment]
Awesome! Well done @sergeyt!
